### PR TITLE
chore(Table): Convert to FC

### PR DESCRIPTION
**Pull request by assuncaocharles**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

# BREAKING CHANGES

This PR converts `Table` component to be functional. Not passing any prop to styles functions.

Related to #12237 

#### Focus areas to test

(optional)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/fluentui/pull/12860)

### CODE REVIEW

File: packages/fluentui/accessibility/src/behaviors/Table/tableBehavior.ts [hunk 1]
```diff
@@ -19,4 +19,8 @@ const tableBehavior: Accessibility = props => ({
   },
 });
 
+export type TableBehaviorProps = {
+  header?: any;
```
Comment (assuncaocharles) [inline packages/fluentui/accessibility/src/behaviors/Table/tableBehavior.ts:23]: Don't know the proper type to put it here.. ? @layershifter @pompomon

File: packages/fluentui/accessibility/src/behaviors/Table/tableBehavior.ts [hunk 2]
```diff
@@ -19,4 +19,8 @@ const tableBehavior: Accessibility = props => ({
   },
 });
 
+export type TableBehaviorProps = {
```
Comment (pompomon) [inline packages/fluentui/accessibility/src/behaviors/Table/tableBehavior.ts:22]: Please why do we need the prop for tableRow in the Table behavior?

Comment (assuncaocharles) [reply packages/fluentui/accessibility/src/behaviors/Table/tableBehavior.ts:22]: It's being used on the `childBehaviors`

File: packages/fluentui/react-northstar/src/components/Table/Table.tsx [hunk 3]
```diff
@@ -90,15 +101,14 @@ class Table extends UIComponent<WithAsProp<TableProps>> {
       return TableRow.create(row, {
         defaultProps: () => ({
           ...props,
-          accessibility: accessibility.childBehaviors ? accessibility.childBehaviors.row : undefined,
+          accessibility: (accessibility && accessibility({ header: !!header })?.childBehaviors.row) || undefined,
```
Comment (pompomon) [inline packages/fluentui/react-northstar/src/components/Table/Table.tsx:104]: Don't forget to add `{...getA11yProps('row', {`

File: packages/fluentui/react-northstar/src/components/Table/Table.tsx [hunk 4]
```diff
@@ -114,40 +124,57 @@ class Table extends UIComponent<WithAsProp<TableProps>> {
     return TableRow.create(header, {
       defaultProps: () => ({
         ...headerRowProps,
-        accessibility: accessibility.childBehaviors ? accessibility.childBehaviors.row : undefined,
+        accessibility: (accessibility && accessibility({ header: !!header })?.childBehaviors.row) || undefined,
```
Comment (pompomon) [inline packages/fluentui/react-northstar/src/components/Table/Table.tsx:127]: Don't forget to add `{...getA11yProps('row', {`

File: packages/fluentui/react-northstar/src/components/Table/Table.tsx [hunk 5]
```diff
@@ -51,34 +58,38 @@ export const tableSlotClassNames: TableSlotClassNames = {
...
-  static displayName = 'Table';
-  static deprecated_className = tableClassName;
-
-  static Cell = TableCell;
-  static Row = TableRow;
-
-  static propTypes = {
-    ...commonPropTypes.createCommon({
-      content: false,
+export type TableStylesProps = Required<Pick<TableProps, 'variables'>>;
```
Comment (pompomon) [inline packages/fluentui/react-northstar/src/components/Table/Table.tsx:61]: I don't think we need to pick variables, @mnajdova @layershifter ?

File: packages/fluentui/react-northstar/src/components/Table/Table.tsx (hunk 4 above)
Comment (assuncaocharles) [reply packages/fluentui/react-northstar/src/components/Table/Table.tsx]: Fixed

File: packages/fluentui/react-northstar/src/components/Table/Table.tsx (hunk 3 above)
Comment (assuncaocharles) [reply packages/fluentui/react-northstar/src/components/Table/Table.tsx]: Fixed

File: packages/fluentui/accessibility/src/behaviors/Table/tableBehavior.ts (hunk 2 above)
Comment (assuncaocharles) [reply packages/fluentui/accessibility/src/behaviors/Table/tableBehavior.ts]: Solved

File: packages/fluentui/react-northstar/src/components/Table/Table.tsx (hunk 5 above)
Comment (layershifter) [reply packages/fluentui/react-northstar/src/components/Table/Table.tsx:61]: Yep, it's true.

```suggestion
export type TableStylesProps = never
```

Comment (assuncaocharles) [reply packages/fluentui/react-northstar/src/components/Table/Table.tsx]: Done :)

**Pull request merged**